zmon-controller-app/src/main/java/org/zalando/zmon/controller/HistoryController.java

```diff
@@ -60,4 +64,16 @@
         return new ResponseEntity<>(historyService.getCheckDefinitionHistory(checkDefinitionId, limit, from, to, action),
                 HttpStatus.OK);
     }
+
+    @RequestMapping(value = "/restoreCheckDefinition", method = RequestMethod.POST)
+    public ResponseEntity<Boolean> restoreCheckDefinition(
+        @RequestBody(required = true) RestoreCheckDefinitionRequest request
+    ) {
+        boolean isRestored = historyService.restoreCheckDefinition(
+                request.getCheckDefinitionHistoryId(),
+                authorityService.getUserName(),
+                new ArrayList<>(authorityService.getTeams()),
+                authorityService.hasAdminAuthority());
+        return new ResponseEntity<>(isRestored, HttpStatus.OK);
```

Comment:
As I understood, result `false` should be in 2 cases: 
 - When the check is not found
 - When there is no permission to change it

Both cases are corner ones, and for me it seems that responding with another response code would be more clear to the client.
I could recommend 204 in case of `true`, and `409 Conflict` otherwise.

Reply:
Totally agree. I would just go with `422 Unprocessable Entity` instead of `409 Conflict`. Because it seems that there is no actual conflict between the request and the state (as RFC suggests for 409); requests is correct but server refuses to process because of two possible reasons.